astropy/wcs/wcsapi/tests/test_fitswcs.py

```diff
@@ -1417,3 +1417,87 @@ def test_pixel_to_world_stokes(wcs_polarized):
     assert isinstance(world[2], StokesCoord)
     assert_array_equal(world[2], [1, 2, 3, 4])
     assert_array_equal(world[2].symbol, ["I", "Q", "U", "V"])
+
+
+@pytest.mark.parametrize("direction", ("world_to_pixel", "pixel_to_world"))
+def test_out_of_bounds(direction):
+    # Make sure that we correctly deal with any out-of-bound values in the
+    # low-level API.
+
+    wcs = WCS(naxis=2)
+    wcs.wcs.crpix = (1, 1)
+    wcs.wcs.set()
+
+    func = (
+        wcs.world_to_pixel_values
+        if direction == "world_to_pixel"
+        else wcs.pixel_to_world_values
+    )
+
+    xp = np.arange(5) + 1
+    yp = np.arange(5) + 1
+
+    # Before setting bounds
+
+    # Scalars
+    xw, yw = func(xp[0], yp[0])
```

Comment:
`xp[0]` returns a type `np.int64`. Can you add a test with Python scalars?

Reply:
Good catch, there was a bug with Python scalars! Fixed now.